src/types/mnemonic.rs

```diff
@@ -39,7 +39,30 @@ impl<'de> Deserialize<'de> for Mnemonic {
             where
                 E: serde::de::Error,
             {
-                BipMnemonic::from_phrase(value, Language::English).map_err(
+                let str_value: String;
+                if value.starts_with("0x") {
+                    // hex value
+                    return Err(serde::de::Error::custom(format!(
+                        "got {} but expected a 12/24 word list ",
+                        value
+                    )));
+                } else if value.starts_with('>') {
+                    todo!("Implement command execution to extract the mnemonic")
+                } else if value.starts_with('$') {
+                    // env
+                    let var = value.strip_prefix('$').unwrap_or(value);
+                    tracing::trace!("Reading {} from env", var);
+                    let val = std::env::var(var).map_err(|e| {
+                        serde::de::Error::custom(format!(
+                            "error while loading this env {}: {}",
+                            var, e,
+                        ))
+                    })?;
+                    str_value = val;
+                } else {
+                    str_value = value.to_string();
+                }
```

Comment:
Here is a better way to write this block:
```suggestion
                let str_value = if value.starts_with("0x") {
                    // hex value
                    return Err(serde::de::Error::custom(format!(
                        "got {} but expected a 12/24 word list ",
                        value
                    )));
                } else if value.starts_with('>') {
                    todo!("Implement command execution to extract the mnemonic")
                } else if value.starts_with('$') {
                    // env
                    let var = value.strip_prefix('$').unwrap_or(value);
                    tracing::trace!("Reading {} from env", var);
                    std::env::var(var).map_err(|e| {
                        serde::de::Error::custom(format!(
                            "error while loading this env {}: {}",
                            var, e,
                        ))
                    })?
                } else {
                    value.to_string()
                };
```